src/v/cluster/producer_state.cc

```diff
@@ -20,6 +20,29 @@ result_promise_t::future_type request::result() const {
     return _result.get_shared_future();
 }
 
+void request::set_value(request_result_t::value_type value) {
+    vassert(
+      _state <= request_state::in_progress && !_result.available(),
+      "unexpected request state during result set: {}",
+      *this);
+    _result.set_value(value);
```

Comment:
std::move? if it matters

Reply:
its an integer under the hood, probably not worth it.